### PR TITLE
ref(toolbar): add flow and fix some comment typos

### PR DESCRIPTION
**Pull request by virtuacoplenny**
Add flow types for files touched by the new toolbox.

Edit: I don't know what I'm doing but I got it to compile. I had the most trouble with AlwaysOnTop and its usage of StatelessToolbarButton configurations as well as VideoQualitySlider and its slider configurations.

### CODE REVIEW

File: react/features/toolbox/components/AbstractToolbarButton.js [hunk 1]
```diff
@@ -1,61 +1,66 @@
...
+     */
+    underlayColor?: ?any
+};
+
+/**
+ * Abstract (base) class for a button in {@link Toolbar}.
+ *
+ * @abstract
+ */
+export default class AbstractToolbarButton<P : Props> extends Component<P> {
```
Comment (saghul) [inline react/features/toolbox/components/AbstractToolbarButton.js]: Wy do you need to alias P to Props? Can't we just use Props like elsewhere?

File: react/features/toolbox/components/StatelessToolbarButton.js [hunk 2]
```diff
@@ -50,35 +51,38 @@ function getButtonAttributesByProps(props: Object = {})
...
- * Represents a button in Toolbar on React.
- *
- * @class ToolbarButton
- * @extends AbstractToolbarButton
+ * The type of the React {@link Component} props of
+ * {@link StatelessToolbarButton}.
  */
-export default class StatelessToolbarButton extends AbstractToolbarButton {
-    _onClick: Function;
+type Props = AbstractToolbarButtonProps & {
```
Comment (saghul) [inline react/features/toolbox/components/StatelessToolbarButton.js]: TIL! :-)

File: react/features/toolbox/components/buttons/AbstractAudioMuteButton.js [hunk 3]
```diff
@@ -14,33 +13,33 @@ import {
...
+    /**
+     * Invoked to toggle audio mute.
+     */
+    dispatch: Dispatch<*>
+};
 
+/**
+ * An abstract implementation of a button for toggling audio mute.
+ */
+export default class AbstractAudioMuteButton<P : Props> extends Component<P> {
```
Comment (saghul) [inline react/features/toolbox/components/buttons/AbstractAudioMuteButton.js]: ditto

File: react/features/toolbox/components/buttons/AbstractHangupButton.js [hunk 4]
```diff
@@ -7,17 +7,29 @@ import {
...
+     * Invoked to toggle audio mute.
+     */
+    dispatch: Dispatch<*>
+};
+
 /**
  * An abstract implementation of a button for leaving the conference.
  */
-export default class AbstractHangupButton extends Component<*> {
+export default class AbstractHangupButton<P : Props> extends Component<P> {
```
Comment (saghul) [inline react/features/toolbox/components/buttons/AbstractHangupButton.js]: ditto

File: react/features/toolbox/components/buttons/AbstractVideoMuteButton.js [hunk 5]
```diff
@@ -14,33 +13,33 @@ import {
...
+    /**
+     * Invoked to toggle video mute.
+     */
+    dispatch: Dispatch<*>
+};
 
+/**
+ * An abstract implementation of a button for toggling video mute.
+ */
+export default class AbstractVideoMuteButton<P : Props> extends Component<P> {
```
Comment (saghul) [inline react/features/toolbox/components/buttons/AbstractVideoMuteButton.js]: ditto

File: react/features/video-quality/components/VideoQualitySlider.web.js [hunk 6]
```diff
@@ -318,25 +312,31 @@ class VideoQualitySlider extends Component {
...
         const { _sliderOptions } = this;
 
         if (_audioOnly) {
-            const audioOnlyOption = _sliderOptions.find(
-                ({ audioOnly }) => audioOnly);
+            const audioOnlyOption: ?SliderOption = _sliderOptions.find(
+                (option: SliderOption) => option.audioOnly);
 
-            return _sliderOptions.indexOf(audioOnlyOption);
+            return audioOnlyOption
```
Comment (saghul) [inline react/features/video-quality/components/VideoQualitySlider.web.js]: can't this be written as concise as before?

File: react/features/toolbox/components/AbstractToolbarButton.js (hunk 1 above)
Comment (virtuacoplenny) [reply react/features/toolbox/components/AbstractToolbarButton.js]: The answer to every question you've asked is: I don't know what I'm doing and I did whatever I could to make flow stop complaining. This specifically is the pattern I saw being used for AbstractDialog and I couldn't find another combination of flow syntax to let the Props be inherited without doing the `P : Props` thing.

File: react/features/video-quality/components/VideoQualitySlider.web.js (hunk 6 above)
Comment (virtuacoplenny) [reply react/features/video-quality/components/VideoQualitySlider.web.js]: Looks like they can be written like they were before. I couldn't make heads or tails of some flow errors and thought this solved one, but looks like the issue was actually solved by the other flow-related changes in the file.

Comment (virtuacoplenny) [reply react/features/video-quality/components/VideoQualitySlider.web.js]: So I think what's happening is in the above flow thinks the .find can return null. And then indexOf shouldn't be called with potentially null and flow doesn't like that. I don't know the flow syntax to get around that.

**Comment by virtuacoplenny** [comment]
Force pushed to resolve conflicts.

**Comment by saghul** [comment]
@virtuacoplenny The changes I'm making will heavily conflict with this. Do you mind holding it for a bit? I'm sorry for the trouble.

**Pull request closed without merging**